src/routes/SafeAppLandingPage/SafeAppLandingPage.tsx

```diff
@@ -0,0 +1,321 @@
+import { ReactElement, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { useLocation } from 'react-router-dom'
+import styled from 'styled-components'
+import { Card, Title, Button, Text, Loader } from '@gnosis.pm/safe-react-components'
+import Divider from '@material-ui/core/Divider'
+import ClickAwayListener from '@material-ui/core/ClickAwayListener'
+import List from '@material-ui/core/List'
+import Popper from '@material-ui/core/Popper'
+
+import SuccessSvg from 'src/assets/icons/safe-created.svg'
+import DemoSvg from 'src/assets/icons/demo.svg'
+import { getChainById } from 'src/config'
+import { getChains } from 'src/config/cache/chains'
+import { generateSafeRoute, history, OPEN_SAFE_ROUTE, SAFE_ROUTES, WELCOME_ROUTE } from 'src/routes/routes'
+import { useAppList } from 'src/routes/safe/components/Apps/hooks/appList/useAppList'
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import { getAppInfoFromUrl } from 'src/routes/safe/components/Apps/utils'
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import { useStateHandler } from 'src/logic/hooks/useStateHandler'
+import { setChainId } from 'src/logic/config/utils'
+import ProviderDisconnected from 'src/components/AppLayout/Header/components/ProviderInfo/ProviderDisconnected'
+import Provider from 'src/components/AppLayout/Header/components/Provider'
+import ConnectDetails from 'src/components/AppLayout/Header/components/ProviderDetails/ConnectDetails'
+import NetworkLabel from 'src/components/NetworkLabel/NetworkLabel'
+import Img from 'src/components/layout/Img'
+import Link from 'src/components/layout/Link'
+
+const DEMO_SAFE_MAINNET = '0xfF501B324DC6d78dC9F983f140B9211c3EdB4dc7'
+
+const demoSafeAppsPath = generateSafeRoute(SAFE_ROUTES.APPS, {
+  shortName: 'eth',
+  safeAddress: DEMO_SAFE_MAINNET,
+})
+
+function SafeAppLandingPage(): ReactElement {
+  const { search } = useLocation()
+  const query = new URLSearchParams(search)
+  const safeAppChainId = query.get('chainId')
+  const safeAppUrl = query.get('appUrl')
+
+  // if no valid chainId or Safe App is present in query params we redirect to the Welcome page
+  useEffect(() => {
+    const isValidChainId = safeAppChainId && getChains().find((chain) => chain.chainId === safeAppChainId)
```

Comment:
You'll need to rename the variable, but we already have `isValidChainId()` in the config file.
```suggestion
    const isValidChainId = isValidChainId(safeAppChainId)
```

Reply:
done!